### PR TITLE
FilesVersions: restore and get contents of versions has been added

### LINKED ISSUE #93: Add support for querying versions of a file
- [x] REQUIRES https://github.com/owncloud/js-owncloud-client/issues/25 to be able to query file id
- [x] oc.fileVersions.listVersions(fileId)
- [ ] oc.fileVersions.getVersionContents(fileId, versionId)
- [ ] oc.fileVersions.restoreVersion(fileId)
- [ ] have an object VersionInfo that extends FileInfo (or whatever makes sense):
    - fileId
    - versionId
    - mtime (timestamp when the version was created)
    - size
    - previewURl

### PR DESCRIPTION
**Pull request by DeepDiver1975**
closes #93

### CODE REVIEW

File: src/fileVersionManagement.js [hunk 1]
```diff
@@ -67,4 +57,61 @@ FilesVersions.prototype.listVersions = function (fileId) {
   })
 }
 
+FilesVersions.prototype.getFileVersionContents = function (fileId, versionId) {
```
Comment (PVince81) [inline src/fileVersionManagement.js]: please add JSDoc

File: src/fileVersionManagement.js [hunk 2]
```diff
@@ -67,4 +57,61 @@ FilesVersions.prototype.listVersions = function (fileId) {
...
+        var err = helpers._parseDAVerror(body)
+        reject(err)
+      }
+    }).catch(error => {
+      reject(error)
+    })
+  })
+}
+
+FilesVersions.prototype.restoreFileVersion = (fileId, versionId, targetPath) => {
```
Comment (PVince81) [inline src/fileVersionManagement.js]: please add JSDoc

File: src/fileVersionManagement.js [hunk 3]
```diff
@@ -67,4 +57,61 @@ FilesVersions.prototype.listVersions = function (fileId) {
...
+  return new Promise((resolve, reject) => {
+    if (!helpers.getAuthorization()) {
```
Comment (PVince81) [inline src/fileVersionManagement.js]: oh, we have a stateful global helper class ?

can we rather encapsulate this into a class and inject it in the constructor ? (separate PR)

**Comment by PVince81** [comment]
well, fix CI and the JSDoc

File: src/fileVersionManagement.js (hunk 3 above)
Comment (DeepDiver1975) [reply src/fileVersionManagement.js]: The helpers Class is injected in the ctor

**Pull request merged**